frontend/src/App.tsx

```diff
@@ -26,6 +27,104 @@ query pinInfo ($after: String) {
 }
 `);
 
+const UPDATE_PIN_STATUS = gql(`
+        mutation updatePinStatus($barcode: String!, $status: PinStatus!) {
+            updateLibraryPinStatus(barcode: $barcode, status: $status) {
+                barcode
+                status
+            }
+        }
+    `);
+
+const pinFragment = gql(`
+fragment pin on LibraryPin {
+  barcode,
+  status
+}
+`)
+
+
+const libraryPinsFragment = gql(`
+fragment pinPage on LibraryPinConnection {
+  
+  edges {
+    cursor
+    node {
+      ...pin
+      loopSize,
+      
+    }
+  }
+}
+
+`)
+
+
+
+export interface UpdatePinStatusProps {
+  item: Record<string, any>
+}
+
+const UpdatePinStatus = ({item }: UpdatePinStatusProps) => {
+
+  const [status, setStatus] = React.useState(item['status']);
+  
+  const handleStatusChange = (event) => {
+      setStatus(event.target.value);
+  }; 
+      const [
+          updateLibraryPinStatus,
+          { error: mutationError }
+        ] = useMutation(UPDATE_PIN_STATUS, {
+
+          update(cache, {data: {updateLibraryPinStatus}}) {
+
+            cache.writeFragment({
+              fragment: pinFragment,
+              data: updateLibraryPinStatus,
+              fragmentName: "pin",
+            });
+    
+            const libraryPins = cache.readFragment({
+              id: "barcode",
+              fragment: libraryPinsFragment,
+              fragmentName: 'pinPage'
+            });
+            if (libraryPins) {    
+              const newEdges = [...libraryPins.edges, updateLibraryPinStatus]
+              cache.writeFragment({
+                id: "barcode",
+                fragment: libraryPinsFragment,
+                fragmentName: 'pinPage',
+                data: { ...libraryPins, edges: newEdges }
+              });
+            }
+          }
+        })
+  
+      return (
+          <div>
+              <form
+                  onSubmit={e => {
+                  e.preventDefault();
+                  updateLibraryPinStatus({ variables: { barcode: item['barcode'], status: status } });
+  
+                  ;
+                  }}
+              >
+                      <Select value={status} onChange={handleStatusChange}>
+                          <option value='READY'>READY</option>
+                          <option value='OCCUPIED'>OCCUPIED</option>
+                          <option value='DIRTY'>DIRTY</option>
+                          <option value='BROKEN'>BROKEN</option>
+                      </Select>
```

Comment:
Could we pull these options from the GraphQL type instead of having to re-list them here?